.github/workflows/ci-basic-build.yml

```diff
@@ -0,0 +1,51 @@
+# This is a boring build workflow that just builds the project on
+# mac, ubuntu and windows. Just to check that the optional compiler
+# flags that don't work on a developer's computer still build on 
+# other OS. 
+name: Boring Build
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**.rs'
+      - '**.toml'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.rs'
+      - '**.toml'
+
+jobs:
+  build:
+    name: Build on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Use rust-toolchain.toml
+        run: rustup show
```

Comment:
This does not work. We have a `rust-toolchain.toml` that sets an MSRV, and it is used automatically; `rust show` only checks what is being used. The `dtolnay/rust-toolchain` action only ensures the specified Rust version is available; you need additional steps to actually use it.